### ISSUE TITLE
Black screen

### ISSUE THREAD

**Issue by Yonle** [bk138/multivnc#188]
**Device**
- Mobile (Android 8.1)
- vivo 1820 (Y91C)

**MultiVNC version**
2.0.7 (Play Store)

**VNC server i'm connecting to**
 - VNC server vendor: tigervnc
 - VNC server version: 1.10.0
 - OS version: 8.1
 - OS language: English

**Describe the bug**
When conndcted into VNC server, I only see a mouse with a black screen.

**To Reproduce**
- Open MultiVNC
- Connect to server

**Expected Behavior**
My VNC Server screen

**Screenshots**
![Screenshot_20220305_150352](https://user-images.githubusercontent.com/63401712/156874567-8eb3e7ad-f507-482c-897b-23d717cdfba9.jpg)

Other VNC client (RealVNC viewer):
![Screenshot_20220305_150430](https://user-images.githubusercontent.com/63401712/156874582-5d37be20-48ef-4281-85fe-d6787b8dc3e6.jpg)

**Comment by bk138** [comment]
Thanks for the report @Yonle ! A few ideas:

* might it be that MultiVNC shows the upper left corner of the remote screen and one has to scroll down to see the other content?
* How would I be able to reproduce your setup? Is it termux?

**Comment by Yonle** [comment]
> * might it be that MultiVNC shows the upper left corner of the remote screen and one has to scroll down to see the other content?
Does not seems that it's there.

> * How would I be able to reproduce your setup? Is it termux?
It's in termux and reproduceable with the following command:

```sh
vncserver -geometry=1280x750 -securitytypes=none -zliblevel=0 :0
```

Can also done with Xvfb + x11vnc:

```sh
Xvfb :0 &
x11vnc --display :0 --forever
```

**Comment by bk138** [comment]
I see. Do you also get a black screen when connecting to another server that's not running on localhost?

**Comment by Yonle** [comment]
> I see. Do you also get a black screen when connecting to another server that's not running on localhost?

Yes. I also tried to connect to my VPS VNC server, unfortunately no luck.

**Comment by bk138** [comment]
> > I see. Do you also get a black screen when connecting to another server that's not running on localhost?
> 
> Yes. I also tried to connect to my VPS VNC server, unfortunately no luck.

Maybe it's the OpenGL drawing. Can you please try with https://github.com/gujjwal00/avnc and report if the issue is there as well (AVNC has similar OpenGL drawing to MultiVNC).

**Comment by Yonle** [comment]
Hi @bk138.

* At this point, AVNC works well as expected. Got no error at it.
* MultiVNC 2.0.0 also result the same black remote screen. Also tried with other VNC server.

**Comment by bk138** [comment]
> Hi @bk138.
> 
>     * At this point, AVNC works well as expected. Got no error at it.
> 
>     * MultiVNC 2.0.0 also result the same black remote screen. Also tried with other VNC server.

No good idea at this point. It seems it's a combination of your device model and MultiVNC. You could try looking at the adb logs and scan for any errors, but that's about it :-/

**Comment by Yonle** [comment]
> > Hi @bk138.
> > 
> >     * At this point, AVNC works well as expected. Got no error at it.
> > 
> >     * MultiVNC 2.0.0 also result the same black remote screen. Also tried with other VNC server.
> 
> No good idea at this point. It seems it's a combination of your device model and MultiVNC. You could try looking at the adb logs and scan for any errors, but that's about it :-/

Well. See whenever i have a chance to get logcat error log.

**Comment by bk138** [comment]
Did a bit of googling:
* This phone's GPU is a PowerVR GE8320 according to https://www.naijatechguide.com/2020/07/vivo-y91c.html
* According to to https://www.notebookcheck.net/PowerVR-GE8320-Graphics-Card-Benchmarks-and-Specs.372646.0.html this is a Series8XE gen which supports OpenGL ES 3.2, which in turn supports `glDrawTexfOES` which MultiVNC uses :thinking:

**Comment by bk138** [comment]
@Yonle you could also [the last version of the 1.8 series](https://apkpure.com/es/multivnc/com.coboltforge.dontmind.multivnc/download/68-APK) give a try. If this works, the issue might be related to the new VNC engine.

**Comment by Yonle** [comment]
> @Yonle you could also [the last version of the 1.8 series](https://apkpure.com/es/multivnc/com.coboltforge.dontmind.multivnc/download/68-APK) give a try. If this works, the issue might be related to the new VNC engine.

And it actually doing it's job well

![Screenshot_20220309_070356](https://user-images.githubusercontent.com/63401712/157346643-b41badea-64f0-41fd-a469-b673b1769e6c.jpg)

**Comment by bk138** [comment]
Interesting! In that case, an adb log from session setup from a recent MultiVNC version probably can shed some light on this if you're interested.
